nengo_viz/static/viz.js

```diff
@@ -127,6 +139,18 @@ VIZ.Component.prototype.on_resize = function(width, height) {};
  */
 VIZ.Component.prototype.on_message = function(event) {};
 
+
+VIZ.Component.prototype.generate_menu = function() {
+    var self = this;
+    var items = [];
+    if (this.label_visible) {
+        items.push(['hide label', function() {self.hide_label();}]);
+    } else {
+        items.push(['show label', function() {self.show_label();}]);    
+    }
+    return items;
```

Comment:
I think by javascript convention, the item in items should be objects as opposed to arrays, then you can refer to them later by accessing the keys of the object as opposed to the indices of the array.

so for example,

items.push({
 title: 'hide label',
 action: function () {
    self.hide_label();
 }
});